src/lib.rs

```diff
@@ -212,6 +262,28 @@ pub fn verify<P: AsRef<[u8]>>(password: P, hash: &str) -> BcryptResult<bool> {
         parts.cost,
         salt.try_into()
             .map_err(|_| BcryptError::InvalidSaltLen(salt_len))?,
+        false,
+    )?;
+    let source_decoded = BASE_64.decode(parts.hash)?;
+    let generated_decoded = BASE_64.decode(generated.hash)?;
+
+    Ok(source_decoded.ct_eq(&generated_decoded).into())
+}
```

Comment:
You should move the inner part into a third function that both verify calls, otherwise we just duplicate the body. Similar to the `_hash_password` in this file

Reply:
Sorry about the sloppy work.

`verify` and `non_truncating_verify` now both call an underlying `_verify` function.